From b5eba1e48af6276e7eec20104a1b79aaf21df485 Mon Sep 17 00:00:00 2001
From: Guilhem IMBERTON <guilhem.imberton@intel.com>
Date: Wed, 29 May 2013 19:11:28 +0200
Subject: Build with optimized gcc-4.7 compiler

BZ: 103828

Use Intel optimized gcc 4.7 to build.

Bug: ADEV-116
Category: Device Enablement
Domain: AOSP-build
Severity: High
Origin: Internal
Upstream-candidate: Yes

Change-Id: I146ad9f5ace5343990a96dba019be3551c2c60e5
Signed-off-by: Guilhem IMBERTON <guilhem.imberton@intel.com>
---
 core/combo/TARGET_linux-x86.mk |   10 +++++++---
 1 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/core/combo/TARGET_linux-x86.mk b/core/combo/TARGET_linux-x86.mk
index 9272d56..ab02902 100644
--- a/core/combo/TARGET_linux-x86.mk
+++ b/core/combo/TARGET_linux-x86.mk
@@ -29,6 +29,8 @@ else
 TARGET_GCC_VERSION := $(TARGET_GCC_VERSION_EXP)
 endif
 
+TARGET_GCC_ARCH ?= x86_64
+
 # Include the arch-variant-specific configuration file.
 # Its role is to define various ARCH_X86_HAVE_XXX feature macros,
 # plus initial values for TARGET_GLOBAL_CFLAGS
@@ -40,11 +42,10 @@ endif
 
 include $(TARGET_ARCH_SPECIFIC_MAKEFILE)
 
-
 # You can set TARGET_TOOLS_PREFIX to get gcc from somewhere else
 ifeq ($(strip $(TARGET_TOOLS_PREFIX)),)
-TARGET_TOOLCHAIN_ROOT := prebuilts/gcc/$(HOST_PREBUILT_TAG)/x86/i686-linux-android-$(TARGET_GCC_VERSION)
-TARGET_TOOLS_PREFIX := $(TARGET_TOOLCHAIN_ROOT)/bin/i686-linux-android-
+TARGET_TOOLCHAIN_ROOT := prebuilts/gcc/$(HOST_PREBUILT_TAG)/x86/$(TARGET_GCC_ARCH)-linux-android-$(TARGET_GCC_VERSION)
+TARGET_TOOLS_PREFIX := $(TARGET_TOOLCHAIN_ROOT)/bin/$(TARGET_GCC_ARCH)-linux-android-
 endif
 
 ifeq ($(TARGET_KERNEL_ARCH),x86_64)
@@ -122,6 +123,9 @@ TARGET_GLOBAL_CPPFLAGS += \
 
 TARGET_GLOBAL_CFLAGS += $(arch_variant_cflags)
 
+# Adding explicit 32 bit flag to avoid gcc auto selection
+TARGET_GLOBAL_CFLAGS += -m32
+
 ifeq ($(ARCH_X86_HAVE_MMX),true)
     TARGET_GLOBAL_CFLAGS += -DUSE_MMX -mmmx
 endif
-- 
1.7.4.1

